### ISSUE TITLE
About Gazebo Simulator

### ISSUE THREAD

**Issue by RyodoTanaka** [Nishida-Lab/forest_robot_project#4]
I already merged the codes for gazebo elements.
But there are some problems we have to resolve.
So, I want to share these info, and need your help.
Following info is same to #3 Pull Request.
1. Now, steer and wheel joints are moved by joint position controller.
   We need to change this controller to velocity controller (We need to make original ros_controller).
2. Not enough at experiment. We need to experiment on step, obstacle, and unevenness terrains.
   Of course, We need to create such terrains on Gazebo (not yet).
3. Sometimes the calculation (simulation) on Gazebo doesn't works well.
   But now, I don't know the reason and solution.

That is all.
Thank you for your cooperation.

**Comment by MoriKen254** [comment]
@RyodoTanaka 

> We need to change this controller to velocity controller (We need to make original ros_controller).

Do you really need to create a velocity controller from scratch?

[ros_control wiki](http://wiki.ros.org/ros_control) says the package also provides `velocity_controllers/joint_velocity_controllers` & `Joint Command Interfaces/Velocity Joint Interface`. 

What about simply replacing the controller and the interface on `Position` to the `Velocity` ones?

**Comment by RyodoTanaka** [comment]
@mum254 

> Do you really need to create a velocity controller from scratch?
> ros_control wiki says the package also provides velocity_controllers/joint_velocity_controllers & Joint Command Interfaces/Velocity Joint Interface.
> What about simply replacing the controller and the interface on Position to the Velocity ones?

Yes, Exactly. But I wanted to say we need a velocity controller that get the `cmd_vel` topic message ( that is `Twist` message I mean). So, I thought we need to calculate  the commands for each joints from  `cmd_vel` message.
And Now, we have a kinematics model and the matrix for calculation. So, I thought we need to create original controller...
Any way, Thank you for your advice and sorry for my less explanation.

**Comment by MoriKen254** [comment]
@RyodoTanaka 
Ooops, excuse me. That makes sense! Surely you need original one =)

**Comment by MoriKen254** [comment]
@RyodoTanaka 
May I try the controller stuff? I'm curious about both the kinematics and ros_control plugin implementation.

@luigicapodieci
I'll try to grasp basics on the paper you introduced to me =).
Let's have a discussion on my next visit (Feb.5).

**Comment by RyodoTanaka** [comment]
@mum254 
I'm sorry to late reply...

Sure, of course ! We need your help.
I have an interest in that and want to join your working.
Let's make it complete !!!

**Comment by MoriKen254** [comment]
@RyodoTanaka 
Thanks!
Yeah. Let's make it together!
